src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIResultProcessingService.java

```diff
@@ -112,9 +108,7 @@ public void removeListener() {
     public void processResult() {
 
         // set lock to prevent multiple nodes from processing the same build job
-        resultQueueLock.lock();
         ResultQueueItem resultQueueItem = resultQueue.poll();
-        resultQueueLock.unlock();
```

Comment:
are we 100% sure this is atomic?

Reply:
The problem with locks is that build agents can potentially crash (e.g. Out of Memory) or be restarted while a lock is active. This already happened in critical situations and then no build agent can continue to work.

In the worst case scenario, this would mean the whole Hazelcast cluster needs to shutdown before the lock is released.

Reducing the lock timeout could help, but in any case, FencedLocks are not supported any more in newer free Hazelcast versions. 

Due to this and other reasons, we also currently evaluate alternatives such as Redis.